assets/js/main.js

```diff
@@ -0,0 +1,5 @@
+"dark"===localStorage.getItem("pref-theme")?document.body.classList.add("dark"):"light"===localStorage.getItem("pref-theme")?document.body.classList.remove("dark"):window.matchMedia("(prefers-color-scheme: dark)").matches&&document.body.classList.add("dark");let menu=document.getElementById("menu");menu&&(menu.scrollLeft=localStorage.getItem("menu-scroll-position"),menu.onscroll=function(){localStorage.setItem("menu-scroll-position",menu.scrollLeft)}),document.querySelectorAll('a[href^="#"]').forEach((e=>{e.addEventListener("click",(function(e){e.preventDefault();var t=this.getAttribute("href").substr(1);window.matchMedia("(prefers-reduced-motion: reduce)").matches?document.querySelector(`[id='${decodeURIComponent(t)}']`).scrollIntoView():document.querySelector(`[id='${decodeURIComponent(t)}']`).scrollIntoView({behavior:"smooth"}),"top"===t?history.replaceState(null,null," "):history.pushState(null,null,`#${t}`)}))}));var mybutton=document.getElementById("top-link");window.onscroll=function(){document.body.scrollTop>800||document.documentElement.scrollTop>800?(mybutton.style.visibility="visible",mybutton.style.opacity="1"):(mybutton.style.visibility="hidden",mybutton.style.opacity="0")},document.getElementById("theme-toggle").addEventListener("click",(()=>{document.body.className.includes("dark")?(document.body.classList.remove("dark"),localStorage.setItem("pref-theme","light")):(document.body.classList.add("dark"),localStorage.setItem("pref-theme","dark"))})),document.querySelectorAll("pre > code").forEach((e=>{const t=e.parentNode.parentNode,o=document.createElement("button");function copyingDone(){o.innerHTML="Kopiert!",setTimeout((()=>{o.innerHTML="Kopieren"}),2e3)}o.classList.add("copy-code"),o.innerHTML="Kopieren",o.addEventListener("click",(t=>{if("clipboard"in navigator)return navigator.clipboard.writeText(e.textContent),void copyingDone();const o=document.createRange();o.selectNodeContents(e);const n=window.getSelection();n.removeAllRanges(),n.addRange(o);try{document.execCommand("copy"),copyingDone()}catch(e){}n.removeRange(o)})),t.classList.contains("highlight")?t.appendChild(o):t.parentNode.firstChild==t||("TABLE"==e.parentNode.parentNode.parentNode.parentNode.parentNode.nodeName?
```

Comment:
minifying js using Hugo would improve readability

Reply:
All right then. Also, would you want to me add  `integrity=sha256.....` for script integrity check?

Reply:
if you want, but it does not make a real difference as the js file is loaded from the same server / domain and not from a CDN.